packages/forms/resources/views/components/date-time-picker.blade.php

```diff
@@ -73,14 +80,17 @@
                 ])
             />
 
-            <span class="absolute inset-y-0 right-0 flex items-center pr-2 pointer-events-none rtl:right-auto rtl:left-0 rtl:pl-2">
-                <svg @class([
-                    'w-5 h-5 text-gray-400',
-                    'dark:text-gray-400' => config('forms.dark_mode'),
-                ]) xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" stroke="currentColor">
-                    <path stroke-linecap="round" stroke-linejoin="round" stroke-width="2" d="M8 7V3m8 4V3m-9 8h10M5 21h14a2 2 0 002-2V7a2 2 0 00-2-2H5a2 2 0 00-2 2v12a2 2 0 002 2z" />
-                </svg>
-            </span>
+            @if ($icon !== false)
+                <span class="absolute inset-y-0 right-0 flex items-center pr-2 cursor-pointer pointer-events-none rtl:right-auto rtl:left-0 rtl:pl-2">
+                    <x-dynamic-component
+                        :component="$icon ?? 'heroicon-s-calendar'"
+                        @class([
+                            'w-5 h-5 text-gray-400 hover:text-gray-800 focus:text-primary-600',
+                            'dark:text-gray-400 dark:hover:text-gray-200 dark:focus:text-primary-600' => config('forms.dark_mode'),
+                        ])
+                    />
+                </span>
```

Comment:
I think we could just use the heroicon instead of the svg we had here.
Could you try refactoring to add the `span` classes onto the `x-dynamic-component` so we can get rid of that wrapping `span` @howdu?

Feel free to share a final screenshot to make sure it all looks good.

Reply:
@zepfietje I think you made these changes in the end - please let me know if not.

Reply:
I did refactor to use dynamic component, but the wrapping span should still be removed.
Could you handle that and move the classes on span to `@class`?

Reply:
I think it's best to keep this separate due the absolute positioning + padding. 

Noticed you've changed the icon to be 'solid' just checking that was intended.

Before
<img width="200" alt="Screenshot 2023-05-18 at 11 08 44" src="https://github.com/filamentphp/filament/assets/533658/cff61723-fda5-4ac8-b3e2-aa68c09929b4">

After
<img width="202" alt="Screenshot 2023-05-18 at 11 07 53" src="https://github.com/filamentphp/filament/assets/533658/6ced08f8-358d-4e1a-8a7a-c057780a1927">

Reply:
Why would the classes have to be on `span` though?

Reply:
@howdu, I've changed the icon back to outline as it would be "breaking".
Will have to update it to be solid/mini in v3.

Reply:
Shouldn't the padding just be swapped out for margin then?

Reply:
Yeah I guess but I always found it not great to mix absolute positioning with margin. Modern browsers handle this ok now but they didn't used to.  

By removing the wrapping span element the CSS will need to be rebuilt to include top: 50% and negative transform because you can’t use flex box to centre the icon. Checking if you think it's still worth it for the one element?